### DISCUSSION TITLE
convert particular pdf to png/jpeg/webp

### DISCUSSION THREAD

**Discussion by theorashid** [ImageMagick/ImageMagick#6271] [Help]
### ImageMagick version

7.1.1-7

### Operating system

MacOS

### Operating system, version and so on

macOS 13.3.1

### Description

I recently updated to v7.1.1-7. I'm converting a bunch of pdfs to webp. There is a particular type of plot that produces a faulty output, where it has missed some points. I used to convert the images using an older version of imagemagick (I can't remember which, I just blindly ran `brew upgrade`) and it worked. For other plots, the conversion to webp has been working fine.

### Steps to Reproduce

The file is test.pdf.

I ran `magick -density 1200 -quality 100 test.pdf test.webp`. And the output is like below – the points in the top-panel plot are missing. I also ran `magick test.pdf test.webp`, `magick test.pdf test.png` `magick test.pdf test.jpeg` and the same issue happened.

I've attached the png rather than the webp as gh don't support webp.

Apologies in advance if this is a newbie mistake, but there a lot of arguments. Also I don't understand why this conversion worked on an older version of imagemagick.

Cheers

### Images

[test.pdf](https://github.com/ImageMagick/ImageMagick/files/11294541/test.pdf)
![test](https://user-images.githubusercontent.com/30835680/233607159-b1f92108-2be7-4011-8191-82af75335e8d.png)

**Comment by snibgo** [comment]
IM delegates rasterization of PDFs to Ghostscript. What version of GS do you have? With GS v10.00, it seems to convert with no problems.

**Reply by theorashid** [reply]
Thanks for getting back. I'm using GS v10.01.1

**Comment by snibgo** [comment]
The points in the top plot are partly transparent. Perhaps that is why they don't appear.

What is the text output from `-verbose`?
```
magick -verbose test.pdf test.webp
```
That may show the GS command used. For example, I get:
```
[ghostscript library 10.00] -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 "-sDEVICE=pngalpha" -dTextAlphaBits=4 -dGraphicsAlphaBits=4 "-r72x72" -dPrinted=false  "-sOutputFile=C:/Users/Alan/AppData/Local/Temp/magick-ljKOi6MKZ-MDR4gFjqiRxbQD47uZlvTn%d" "-fC:/Users/Alan/AppData/Local/Temp/magick-sUjWRBXer8Mq_AGreQeeZKCAWVpZDEUe" "-fC:/Users/Alan/AppData/Local/Temp/magick-NWPHSQglVqXALW83NWWha5h19KmjRk_V"C:/Users/Alan/AppData/Local/Temp/magick-ljKOi6MKZ-MDR4gFjqiRxbQD47uZlvTn1 PNG 267x504 267x504+0+0 8-bit TrueColorAlpha sRGB 38464B 0.000u 0:00.003
test.pdf PDF 267x504 267x504+0+0 16-bit TrueColorAlpha sRGB 38464B 0.000u 0:00.001
test.pdf=>t.webp PDF 267x504 267x504+0+0 16-bit TrueColorAlpha sRGB 12686B 0.016u 0:00.015
```

**Reply by theorashid** [reply]
I've converted other plots with transparent points and not had the same issue. Looking at the output again, it's quite weird that one point has appeared but not the rest. Here's the output with `-verbose`:

```
magick -verbose  test.pdf test.webp
/var/folders/vj/3hks9tp1269_yz2d30q5xfrc0000gn/T/magick-NVUYdctXGFQ4mgzsyrzoUuew2RCs6hX51 PNG 267x504 267x504+0+0 8-bit TrueColorAlpha sRGB 32067B 0.000u 0:00.001
test.pdf PDF 267x504 267x504+0+0 16-bit TrueColorAlpha sRGB 32067B 0.000u 0:00.000
[ghostscript library 10.01] -sstdout=%stderr -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 '-sDEVICE=pngalpha' -dTextAlphaBits=4 -dGraphicsAlphaBits=4 '-r72x72' -dPrinted=false  '-sOutputFile=/var/folders/vj/3hks9tp1269_yz2d30q5xfrc0000gn/T/magick-NVUYdctXGFQ4mgzsyrzoUuew2RCs6hX5%d' '-f/var/folders/vj/3hks9tp1269_yz2d30q5xfrc0000gn/T/magick-44aFYcGel07xpIA4IUPGYzOt5U8FnYm9' '-f/var/folders/vj/3hks9tp1269_yz2d30q5xfrc0000gn/T/magick-kik3zhOpSp9DwT4gYEu7NV_uGU4Tfg7P'test.pdf=>test.webp PDF 267x504 267x504+0+0 16-bit TrueColorAlpha sRGB 8394B 0.020u 0:00.016
```

**Comment by snibgo** [comment]
I can't see a problem with that GS command. I thought perhaps `-sDEVICE=pngalpha` would be wrong, but it isn't.

Try directly running Ghostscript, like this, but your path to GS will be different (I am on Windows):

```
C:\ProgramFiles\gs\gs10.00.0\bin\gswin64c -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 '-sDEVICE=pngalpha' -dTextAlphaBits=4 -dGraphicsAlphaBits=4 '-r72x72' -dPrinted=false  '-sOutputFile=x.png' test.pdf
```

If that works, then I suspect a problem within IM. Try saving to PNG instead of WEBP.

If it doesn't work, then the problem is in GS.

**Comment by theorashid** [comment]
I ran `gs -q -dQUIET -dSAFER -dBATCH -dNOPAUSE -dNOPROMPT -dMaxBitmap=500000000 -dAlignToPixels=0 -dGridFitTT=2 '-sDEVICE=pngalpha' -dTextAlphaBits=4 -dGraphicsAlphaBits=4 '-r72x72' -dPrinted=false  '-sOutputFile=x.png' test.pdf`

And got the same (incorrect) output. So it must be a problem with ghostscipt. What are the next steps here? It looks like v10.01.1 deprecated the [old pdf interpreter and some other args](https://ghostscript.readthedocs.io/en/latest/News.html), so it makes sense that this is causing an issue. I can't find anywhere on GH to contact them

**Comment by snibgo** [comment]
It seems that Ghostscript bugs should be reported at https://bugs.ghostscript.com/

**Comment by theorashid** [comment]
FYI, this has been fixed here https://bugs.ghostscript.com/show_bug.cgi?id=706643

So hopefully it will all work with the next release of Ghostscript
